### PR TITLE
blog assignment code review

### PR DESCRIPTION
**Pull request by tkbeili**
Code review for the blog assignment delivery. This PR includes code written from June 7th until July 12th

### CODE REVIEW

File: app/controllers/favourites_controller.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      redirect_to blog_path(@post), notice: "Added favourite"
+    else
+      redirect_to blog_path(@post), notice: "Already added"
+    end
+  end
+
+  def destroy
+    @post = Post.find params[:post_id]
+    @favourite = Favourite.find params[:id]
+    @favourite.destroy if can? :destroy, Favourite
```
Comment (tkbeili) [inline app/controllers/favourites_controller.rb:20]: This should probably be `if can? :destroy, @favourite`

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
+class PostsController < ApplicationController
+  before_action :authenticate_user!, except: [:show, :index]
+  def authenticate_user!
+    redirect_to new_session_path, alert: "Please sign in" unless user_signed_in?
+  end
+  def posts
+    # @posts = Post.order("created_at")
```
Comment (tkbeili) [inline app/controllers/posts_controller.rb]: Please remove commented out code

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    @comment = Comment.new
+  end
+
+  def show
+    @comment = Comment.find params[:id]
+  end
+
+  def create
+    @post = Post.find params[:post_id]
+    comment_params = params.require(:comment).permit(:body)
```
Comment (leehyangnim) [inline app/controllers/comments_controller.rb:20]: refactoring - better to make a private method called comment_params

File: app/views/users/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1,33 @@
+<h1>Edit Info</h1>
+<% if flash[:notice] %>
+  <div class="flash-notice"><%= flash[:notice] %></div>
+<% elsif flash[:alert] %>
+  <div class="flash-alert"><%= flash[:alert] %></div>
+<% end %>
+<%= form_tag user_path, method: :patch do %>
```
Comment (leehyangnim) [inline app/views/users/edit.html.erb:7]: maybe gem simple form?

File: config/routes.rb [hunk 5]
```diff
@@ -1,4 +1,33 @@
 Rails.application.routes.draw do
   get "/about" => "home#about"
   root "home#index"
+
+  get "/posts" => "posts#posts"
+
+  get "/home/new" => "posts#new", as: :new_post
```
Comment (leehyangnim) [inline config/routes.rb:7]: don't need it since you already have resources :posts

**Pull request merged**